libheif/context.cc

```diff
@@ -609,6 +609,17 @@ Error HeifContext::interpret_heif_file()
         uint32_t width = ispe->get_width();
         uint32_t height = ispe->get_height();
 
+        uint32_t max_width_height = static_cast<uint32_t>(std::numeric_limits<int>::max());
+        if (width >= max_width_height || height >= max_width_height) {
+          std::stringstream sstr;
+          sstr << "Image size " << width << "x" << height << " exceeds the maximum image size "
+                << m_maximum_image_size_limit << "\n";
+
+          return Error(heif_error_Memory_allocation_error,
+                        heif_suberror_Security_limit_exceeded,
+                        sstr.str());
+        }
+
         image->set_resolution(width, height);
```

Comment:
Simpler fix would be to set the argument type for `set_resolution()` to `uint32_t`. It is `uint32_t` internally anyway.

Reply:
Ok, but it's returned as `int` in the getters and I didn't want to introduce any potential other errors by changing the type in various places.
https://github.com/strukturag/libheif/blob/e42b01f3b46eb02545b0ddeff4368d3dd3bdb430/libheif/context.h#L117-L119

Also it's returned as `int` in the public API:
https://github.com/strukturag/libheif/blob/e42b01f3b46eb02545b0ddeff4368d3dd3bdb430/libheif/api/libheif/heif.cc#L804
https://github.com/strukturag/libheif/blob/e42b01f3b46eb02545b0ddeff4368d3dd3bdb430/libheif/api/libheif/heif.cc#L815

Reply:
Ok, makes sense.